youtube/feed.go

```diff
@@ -276,39 +277,36 @@ var (
 	ErrMaxCustomMessageLength = errors.New("max length of custom message can be 500 chars")
 )
 
-func (p *Plugin) parseYtUrl(url string) (t ytUrlType, id string, err error) {
-	if ytUrlShortRegex.MatchString(url) {
-		capturingGroups := ytUrlShortRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][2]) > 0 {
-			return ytUrlTypeVideo, capturingGroups[0][2], nil
-		}
-	} else if ytVideoUrlRegex.MatchString(url) {
-		capturingGroups := ytVideoUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][4]) > 0 {
-			return ytUrlTypeVideo, capturingGroups[0][4], nil
-		}
-	} else if ytChannelUrlRegex.MatchString(url) {
-		capturingGroups := ytChannelUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeChannel, capturingGroups[0][5], nil
-		}
-	} else if ytCustomUrlRegex.MatchString(url) {
-		capturingGroups := ytCustomUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeCustom, capturingGroups[0][5], nil
-		}
-	} else if ytUserUrlRegex.MatchString(url) {
-		capturingGroups := ytUserUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeUser, capturingGroups[0][5], nil
-		}
-	} else if ytHandleUrlRegex.MatchString(url) {
-		capturingGroups := ytHandleUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeHandle, capturingGroups[0][5], nil
-		}
+func (p *Plugin) parseYtUrl(url string) (idType ytUrlType, id string, err error) {
+	var valid bool
+	if id, valid = p.getIDFromRegexUrl(ytUrlShortRegex, url, 2); valid {
+		idType = ytUrlTypeVideo
+	} else if id, valid = p.getIDFromRegexUrl(ytVideoUrlRegex, url, 4); valid {
+		idType = ytUrlTypeVideo
+	} else if id, valid = p.getIDFromRegexUrl(ytChannelUrlRegex, url, 5); valid {
+		idType = ytUrlTypeChannel
+	} else if id, valid = p.getIDFromRegexUrl(ytCustomUrlRegex, url, 5); valid {
+		idType = ytUrlTypeCustom
+	} else if id, valid = p.getIDFromRegexUrl(ytUserUrlRegex, url, 5); valid {
+		idType = ytUrlTypeUser
+	} else if id, valid = p.getIDFromRegexUrl(ytHandleUrlRegex, url, 5); valid {
+		idType = ytUrlTypeHandle
+	} else {
+		idType = ytUrlTypeInvalid
+		err = errors.New("invalid or incomplete url")
 	}
-	return ytUrlTypeInvalid, "", errors.New("invalid or incomplete url")
+
+	return idType, id, err
+}
+
+func (p *Plugin) getIDFromRegexUrl(expr *regexp.Regexp, url string, index int) (id string, valid bool) {
```

Comment:
```suggestion
func (p *Plugin) getIDFromRegexUrl(expr *regexp.Regexp, url string, index int) (id string, ok bool) {
```
`ok` is the canonical name for return values of this type.

Alternately, you can omit the return value names entirely; they're sufficiently clear in context.